1.16.5/src/main/java/io/github/axolotlclient/modules/hypixel/autogg/AutoGG.java

```diff
@@ -158,28 +161,28 @@ private List<String> addToList(String... strings) {
 		return Arrays.stream(strings).collect(Collectors.toList());
 	}
 
-	public void onMessage(Text message) {
+	public void onMessage(ReceiveChatMessageEvent event) {
 		if (client.getCurrentServerEntry() != null) {
 			serverMap.keySet().forEach(s -> {
 				if (serverMap.get(s).get() && client.getCurrentServerEntry().address.contains(s)) {
 					if (gf.get()) {
-						processChat(message, gfStrings.get(s), gfString.get());
+						processChat(event.getOriginalMessage(), gfStrings.get(s), gfString.get());
 					}
 					if (gg.get()) {
-						processChat(message, ggStrings.get(s), ggString.get());
+						processChat(event.getOriginalMessage(), ggStrings.get(s), ggString.get());
 					}
 					if (glhf.get()) {
-						processChat(message, glhfStrings.get(s), glhfString.get());
+						processChat(event.getOriginalMessage(), glhfStrings.get(s), glhfString.get());
 					}
 				}
 			});
 		}
 	}
 
-	private void processChat(Text messageReceived, List<String> options, String messageToSend) {
+	private void processChat(String messageReceived, List<String> options, String messageToSend) {
 		if (System.currentTimeMillis() - this.lastTime > 3000 && options != null) {
 			for (String s : options) {
-				if (messageReceived.getString().contains(s)) {
+				if (messageReceived.contains(s)) {
 					Util.sendChatMessage(messageToSend);
```

Comment:
If on hypixel this needs to prepend `/achat ` so it will always send in public chat and not in party/guild chat.

Reply:
since the string sent in the chat is only defined in an option this is user-configurable and can be configured to suit whatever purpose the user intends.